tests/ui/byte_char_slice.stderr

```diff
@@ -6,5 +6,23 @@ LL |     let bad = &[b'a', b'b', b'c'];
    |
    = note: `-D clippy::byte-char-slice` implied by `-D warnings`
 
-error: aborting due to previous error
+error: can be more succinctly written as a byte str
+  --> $DIR/byte_char_slice.rs:7:18
+   |
+LL |     let quotes = &[b'"', b'H', b'i'];
+   |                  ^^^^^^^^^^^^^^^^^^^ help: try: `b"/"Hi"`
+
+error: can be more succinctly written as a byte str
+  --> $DIR/byte_char_slice.rs:8:18
+   |
+LL |     let quotes = &[b'/'', b'S', b'u', b'p'];
+   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `b"'Sup"`
+
+error: can be more succinctly written as a byte str
+  --> $DIR/byte_char_slice.rs:9:19
+   |
+LL |     let escapes = &[b'/x42', b'E', b's', b'c'];
```

Comment:
issue: wrong kind of slash!

Reply:
Could that be a compiletest artifact? The `.fixed` version looks OK.

Reply:
hmmmmm yeah probably!

feels like a bug

Reply:
Yeah I was wondering about this as well, but it does the right thing anyway, so I don't know what I could do

Reply:
Having reviewed compiletest code, I'm now pretty sure this is a slash normalization to ease dealing with paths across windows/unices.